### PR TITLE
Delete baseline pipeline classes

### LINKED ISSUE #2175: Delete baseline pipeline classes
From the discussion on #2091 and suggestion from @freddyaboulton and @dsherry, we should delete the baseline classes as they're only being used in AutoMLSearch. We can replace them with utility methods instead.

https://github.com/alteryx/evalml/pull/2091#discussion_r617927899

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #2175

Also fixes minor bug with getting input feature names for baseline / pipelines with just one component :)

To discuss:
- Still keep tests as automl tests? I decided to add some tests in automl to make sure that we're getting the right predictions.

### CODE REVIEW

File: evalml/automl/automl_search.py [hunk 1]
```diff
@@ -648,25 +648,38 @@ def _validate_problem_type(self):
...
             gap = self.problem_configuration['gap']
             max_delay = self.problem_configuration['max_delay']
-            baseline = pipeline_class(parameters={"pipeline": {"gap": gap, "max_delay": max_delay},
+            baseline = pipeline_class(component_graph=["Time Series Baseline Estimator"],
+                                      custom_name=pipeline_name,
+                                      parameters={"pipeline": {"gap": gap, "max_delay": max_delay},
                                                   "Time Series Baseline Estimator": {"gap": gap, "max_delay": max_delay}})
+        return baseline
+
+    def _add_baseline_pipelines(self):
```
Comment (angela97lin) [inline evalml/automl/automl_search.py:677]: Just separated out helpers to make logic of getting baseline abstracted from training baseline :) No functional change.

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -23,6 +24,7 @@ Release Notes
 .. warning::
 
     **Breaking Changes**
+        * All baseline pipeline classes (``BaselineBinaryPipeline``, ``BaselineMulticlassPipeline``, ``BaselineRegressionPipeline``, etc.) have been deleted :pr:`2202`
```
Comment (bchen1116) [inline docs/source/release_notes.rst:27]: 👌

**Review by bchen1116 — APPROVED** [review]
Nice changes! LGTM

**Review by freddyaboulton — APPROVED** [review]
Looks great @angela97lin ! Agree it's a good idea to add tests for the baseline predictions in automl.

File: evalml/pipelines/component_graph.py [hunk 3]
```diff
@@ -162,7 +162,9 @@ def _fit_transform_features_helper(self, needs_fitting, X, y=None):
             ww.DataTable: Transformed values.
         """
         if len(self.compute_order) <= 1:
-            return infer_feature_types(X)
+            X = infer_feature_types(X)
+            self.input_feature_names.update({self.compute_order[0]: list(X.columns)})
```
Comment (dsherry) [inline evalml/pipelines/component_graph.py:166]: Good catch

File: evalml/tests/automl_tests/test_automl.py [hunk 4]
```diff
@@ -1372,11 +1377,11 @@ def test_percent_better_than_baseline_computed_for_all_objectives(mock_time_seri
...
-    baseline_pipeline_class = {"binary": "evalml.pipelines.ModeBaselineBinaryPipeline",
-                               "multiclass": "evalml.pipelines.ModeBaselineMulticlassPipeline",
-                               "regression": "evalml.pipelines.MeanBaselineRegressionPipeline",
-                               "time series regression": "evalml.pipelines.TimeSeriesBaselineRegressionPipeline"
-                               }[problem_type]
+    baseline_pipeline_class = {ProblemTypes.BINARY: "evalml.pipelines.BinaryClassificationPipeline",
+                               ProblemTypes.MULTICLASS: "evalml.pipelines.MulticlassClassificationPipeline",
+                               ProblemTypes.REGRESSION: "evalml.pipelines.RegressionPipeline",
+                               ProblemTypes.TIME_SERIES_REGRESSION: "evalml.pipelines.TimeSeriesRegressionPipeline"
+                               }[problem_type_enum]
```
Comment (dsherry) [inline evalml/tests/automl_tests/test_automl.py:1384]: Enums are great :)

File: evalml/tests/automl_tests/test_automl.py [hunk 5]
```diff
@@ -1463,6 +1464,8 @@ def new(self, parameters, random_seed=0):
 
         def clone(self):
             return self.__class__(self.parameters, random_seed=self.random_seed)
+    mock_score = MagicMock(side_effect=[{"Log Loss Binary": 1, "F1": val} for val in fold_scores])
+    DummyPipeline.score = mock_score
```
Comment (dsherry) [inline evalml/tests/automl_tests/test_automl.py:1468]: Why is this change necassary? I thought the old mocking would work just fine.

**Review by dsherry — APPROVED** [review]
🚢 😁

File: evalml/tests/automl_tests/test_automl.py (hunk 5 above)
Comment (angela97lin) [reply evalml/tests/automl_tests/test_automl.py:1468]: Previously, we separated out and mocked "evalml.pipelines.ModeBaselineBinaryPipeline.score" differently from DummyPipeline. Now, we no longer have ModeBaselineBinaryPipeline to mock, so I'm mocking the first pipeline (baseline) via the BinaryClassificationPipeline class and then mocking this custom class with this line.

**Pull request merged**